contract/src/test/java/io/neow3j/contract/Nep5TokenTest.java

```diff
@@ -141,24 +183,69 @@ public void testGetBalanceOfWallet() throws Exception {
                 is(new BigInteger("600000000")));
     }
 
-    @Test(expected = InsufficientFundsException.class)
+    @Test
     public void testFailTransferFromDefaultAccount_InsufficientBalance() throws Exception {
-        setUpWireMockForInvokeFunction("decimals", "invokefunction_decimals_gas.json");
         setUpWireMockForGetBlockCount(1000);
         setUpWireMockForBalanceOf(account1.getScriptHash(),
                 "invokefunction_balanceOf_300000000.json");
         setUpWireMockForCall("invokescript", "invokescript_transfer.json");
 
+        exceptionRule.expect(InsufficientFundsException.class);
+        exceptionRule.expectMessage("default account does not hold enough tokens");
         gasToken.transferFromDefaultAccount(Wallet.withAccounts(account1), RECIPIENT_SCRIPT_HASH,
                 new BigDecimal("4"));
     }
 
-    @Test(expected = IllegalArgumentException.class)
+    @Test
     public void testTransferFromDefaultAccount_negativeAmount() throws IOException {
+        exceptionRule.expect(IllegalArgumentException.class);
+        exceptionRule.expectMessage("amount must be greater than or equal to 0");
         neoToken.transferFromDefaultAccount(Wallet.withAccounts(account1), RECIPIENT_SCRIPT_HASH,
                 new BigDecimal("-1"));
     }
 
+    @Test
+    public void testTransferInvalidDecimalsInAmount() throws Throwable {
+        exceptionRule.expect(IllegalArgumentException.class);
+        exceptionRule.expectMessage("amount contains more decimal places than this token can handle");
+        neoToken.transferFromDefaultAccount(Wallet.withAccounts(account1),
+                RECIPIENT_SCRIPT_HASH, new BigDecimal("0.1"));
+    }
+
+    @Test
+    public void testTransferInvalidDecimalsInAmount_trailingZeros() throws IOException {
+        setUpWireMockForBalanceOf(account1.getScriptHash(),
+                "invokefunction_balanceOf_300000000.json");
+
+        // Trailing zeros should be ignored - this should not throw an exception.
```

Comment:
Add a try-catch and put a `fail()` into the catch block. That makes it more clear what the intention of the test is.

Reply:
Good point, I added a try-catch for that.